tests/status.py

```diff
@@ -72,6 +72,62 @@ def runTest(self):
         board.expect(prompt)
         self.result_message = 'ifconfig shows ip addresses: %s' % tmp
 
+class MacAddressCheck(rootfs_boot.RootFSBootTest):
+    '''Checked MAC addresses for all interfaces.'''
+    def __checkMac(self, num1, num2, count):
+        if num1:
+            num1 = re.sub('[:]', '', num1.upper())[-6:]
+            result = int(num2, base=16) + count
+            if result == int(num1, base=16):
+                self.result_message = 'MAC addresses are sequential'
+            else:
+                raise ValueError('MAC addresses are not sequential')
+        else:
+            raise ValueError('Could not find appropriate MAC address in %s' % num1)
+
+    def runTest(self):
+#       TODO: enable wi-fi mac address check from /sys/class once wi-fi tests are added
+#       board.sendline('\ncat /sys/class/net/wlan0/address')
+#       board.expect('wlan0/address')
+#       board.expect(prompt)
+#       mac0 = re.search(r'00:19:f5:([a-z0-9:]{8})', board.before).group()
+#       mac0 = re.sub('[:]', '', mac0.upper())[-6:]
+#       board.sendline('\ncat /sys/class/net/wlan0-1/address')
+#       board.expect('wlan0-1/address')
+#       board.expect(prompt)
+#       mac1 = re.search(r'00:19:f5:([a-z0-9:]{8})', board.before).group()
+#       self.__checkMac(mac1, mac0, 1)
+        board.sendline('\nhexdump -n 6 -v -e \'8/1 "%02X" "\n"\' /proc/device-tree/uccp@18480000/mac-address0')
+        board.expect('mac-address0')
+        board.expect(prompt)
+        mac0 = re.search(r'0019F5([A-Z0-9]{6})', board.before).group()
+        mac0 = mac0.upper()[-6:]
+        board.sendline('\nhexdump -n 6 -v -e \'8/1 "%02X" "\n"\' /proc/device-tree/uccp@18480000/mac-address1')
+        board.expect('mac-address1')
+        board.expect(prompt)
+        mac1 = re.search(r'0019F5([A-Z0-9]{6})', board.before).group()
+        self.__checkMac(mac1, mac0, 1)
+        board.sendline('\nhcitool dev')
+        board.expect('hci0')
+        board.expect(prompt)
+        mac2 = re.search(r'00:19:F5:([A-Z0-9:]{8})', board.before).group()
+        self.__checkMac(mac2, mac0, 2)
+        board.sendline('\ncat /sys/class/net/eth0/address')
+        board.expect('eth0/address')
+        board.expect(prompt)
+        mac3 = re.search(r'00:19:f5:([a-z0-9:]{8})', board.before).group()
+        self.__checkMac(mac3, mac0, 3)
+        board.sendline('\ncat /sys/class/net/lowpan0/address')
+        board.expect('lowpan0/address')
+        board.expect(prompt)
+        mac5 = re.search(r'02:19:f5:ff:fe:([a-z0-9:]{8})', board.before).group()
+        self.__checkMac(mac5, mac0, 5)
+        board.sendline('\ncat /sys/class/net/wpan0/address')
+        board.expect('wpan0/address')
+        board.expect(prompt)
+        mac5 = re.search(r'02:19:f5:ff:fe:([a-z0-9:]{8})', board.before).group()
+        self.__checkMac(mac5, mac0, 5)
```

Comment:
I would keep the current set of commands as it is for following reasons:

1. The code would fail pep8 and pylint check for such a lengthy function call.  And even if some part is pushed to new lines, it wouldn't make the function use or the code significantly shorter e.g.

        self._checkMac("\nhexdump -n 6 -v -e \'8/1 "%02X" "\n"\' /proc/device-tree/uccp@18480000/mac-address1",
        'mac-address0', '0019F5([A-Z0-9]{2}){3}', mac0, 1)

1. All other tests use the same set of commands (e.g. uname.py, other tests in status.py).  So, although we can start to add more concise and generic code from now, but this would lead to inconsistency and affect code readability